infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/receive/PersistedWriterMetadata.java

```diff
@@ -0,0 +1,268 @@
+
+        return;
+    }
+
+
+    /**
+     * This call should be done in a transaction after a transfer done and before apply the snapshot.
+     * @param ts
+     */
+    public void setLastSnapTransferDoneTimestamp(long siteConfigID, long ts) {
+        CorfuStoreMetadata.Timestamp timestamp = corfuStore.getTimestamp();
+        long persisteSiteConfigID = query(timestamp, PersistedWriterMetadataType.SiteConfigID);
+        long persistSnapStart = query(timestamp, PersistedWriterMetadataType.LastSnapStart);
+
+        log.debug("setLastSnapTransferDone snapshotStart siteConfigID " + siteConfigID + " ts " + ts +
+                " persisteSiteConfigID " + persisteSiteConfigID + " persistSnapStart " + persistSnapStart);
+
+        // It means the site config has changed, ingore the update operation.
+        if (siteConfigID != persisteSiteConfigID || ts <= persisteSiteConfigID) {
+            log.warn("The metadata is older than the presisted one. Set snapshotStart siteConfigID " + siteConfigID + " ts " + ts +
+                    " persisteSiteConfigID " + persisteSiteConfigID + " persistSnapStart " + persistSnapStart);
+            return;
+        }
+
+        TxBuilder txBuilder = corfuStore.tx(namespace);
+
+        //Update the siteConfigID to fence all other transactions that update the metadata at the same time
+        appendUpdate(txBuilder, PersistedWriterMetadataType.SiteConfigID, siteConfigID);
+
+        //Setup the LastSnapStart
+        appendUpdate(txBuilder, PersistedWriterMetadataType.LastSnapTransferDone, ts);
+
+        txBuilder.commit(timestamp);
+
+        log.debug("Commit. Set snapshotStart siteConfigID " + siteConfigID + " ts " + ts +
+                " persisteSiteConfigID " + persisteSiteConfigID + " persistSnapStart " + persistSnapStart);
+        return;
+    }
+
+    public void setSrcBaseSnapshotDone(LogReplicationEntry entry) {
+        CorfuStoreMetadata.Timestamp timestamp = corfuStore.getTimestamp();
+        long persistSiteConfigID = query(timestamp, PersistedWriterMetadataType.SiteConfigID);
+        long persistSnapStart = query(timestamp, PersistedWriterMetadataType.LastSnapStart);
+        long persistSnapTranferDone = query(timestamp, PersistedWriterMetadataType.LastSnapTransferDone);
+        long siteConfigID = entry.getMetadata().getSiteConfigID();
+        long ts = entry.getMetadata().getSnapshotTimestamp();
+
+        if (siteConfigID != persistSiteConfigID || ts != persistSnapStart || ts != persistSnapTranferDone) {
+            log.warn("siteConfigID " + siteConfigID + " != " + " persist " + persistSiteConfigID +  " ts " + ts +
+                    " != " + "persistSnapTranferDone " + persistSnapTranferDone);
+            return;
+        }
+
+        TxBuilder txBuilder = corfuStore.tx(namespace);
+
+        //Update the siteConfigID to fence all other transactions that update the metadata at the same time
```

Comment:
what is this fencing that is done ? Is it like all operations update the siteConfigId so they fence each other ? Why do you need this ?